yargs.js

```diff
@@ -443,8 +438,8 @@ function Yargs (processArgs, cwd, parentRequire) {
     var parsed = parseArgs(args, shortCircuit)
     if (parseFn) {
       parseFn(exitError, parsed, output)
+      self.reset()
```

Comment:
@bcoe This is a lot more (unnecessary) overhead than just calling `resetForNextParse()`, meaning `reset()` will be called N+1 times (where N is the number of commands) for `yargs.parse(msg, cb)`.